lib/multigrid.cpp

```diff
@@ -1591,22 +1602,156 @@ namespace quda
 
     // reenable Schwarz
     if (schwarz_reset) {
-      if (getVerbosity() >= QUDA_VERBOSE) printfQuda("Reenabling Schwarz for null-space finding");
+      logQuda(QUDA_VERBOSE, "Reenabling Schwarz for null-space finding");
       int commDim[QUDA_MAX_DIM];
       for (int i=0; i<QUDA_MAX_DIM; i++) commDim[i] = 0;
       diracSmootherSloppy->setCommDim(commDim);
     }
 
-    if (param.mg_global.vec_store[param.level] == QUDA_BOOLEAN_TRUE) { // conditional store of null vectors
-      saveVectors(B);
+    popLevel();
+  }
+
+  void MG::generateChebyshevFilter(std::vector<ColorSpinorField *> &B)
```

Comment:
Is there any reason one might think that the Chebyshev filter would be better placed in a function or method outside of the MG class?  Just wondering if it could be helpful elsewhere, and / or to reduce redundant code elsewhere.

Reply:
I think I would need to make the existing Chebyshev polynomial routine (which gets reused across `CA-CG` and `CA-GCR`) more modular/featureful. More specifically, I'd need to expose the periodic rescaling (line 1687-ish), plus saving vectors along the way (loop on line 1719-ish). I can think that through.

Reply:
https://github.com/lattice/quda/pull/1283/commits/b4cdb6f76184f592017bd8f43428dd875d45b02d